# HealthCare Analysis

## EDA

Libraries and Data

In [1]:
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import re


In [2]:
print("You are here:",os.getcwdb())

You are here: b'/Users/amithmg/Documents/GitHub/nlp_aiwp_3129'


In [3]:
!dir

zsh:1: command not found: dir


In [4]:
hc_df = pd.read_csv('dataset/kaggle healthcare_dataset.csv')

### Data description and detailing (READMe)

* Health care data has 55500 records and 15 variables
* Patient details are shared in this data such as name, age, gender, medication, doctor and more.
* Data types(dtypes): float64(1), int64(2), object(12).
* Data types must be changed for variables listed below:
    * Discharge Date: object to Date
    * Date of Admission: object to Date




In [5]:
hc_df.shape

(55500, 15)

In [6]:
hc_df.columns

Index(['Name', 'Age', 'Gender', 'Blood Type', 'Medical Condition',
       'Date of Admission', 'Doctor', 'Hospital', 'Insurance Provider',
       'Billing Amount', 'Room Number', 'Admission Type', 'Discharge Date',
       'Medication', 'Test Results'],
      dtype='object')

In [7]:
hc_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Name                55500 non-null  object 
 1   Age                 55500 non-null  int64  
 2   Gender              55500 non-null  object 
 3   Blood Type          55500 non-null  object 
 4   Medical Condition   55500 non-null  object 
 5   Date of Admission   55500 non-null  object 
 6   Doctor              55500 non-null  object 
 7   Hospital            55500 non-null  object 
 8   Insurance Provider  55500 non-null  object 
 9   Billing Amount      55500 non-null  float64
 10  Room Number         55500 non-null  int64  
 11  Admission Type      55500 non-null  object 
 12  Discharge Date      55500 non-null  object 
 13  Medication          55500 non-null  object 
 14  Test Results        55500 non-null  object 
dtypes: float64(1), int64(2), object(12)
memory usage: 6.4

In [8]:
hc_df.head(10)

Name  Age  Gender Blood Type Medical Condition  \
0       Bobby JacksOn   30    Male         B-            Cancer   
1        LesLie TErRy   62    Male         A+           Obesity   
2         DaNnY sMitH   76  Female         A-           Obesity   
3        andrEw waTtS   28  Female         O+          Diabetes   
4       adrIENNE bEll   43  Female        AB+            Cancer   
5       EMILY JOHNSOn   36    Male         A+            Asthma   
6      edwArD EDWaRDs   21  Female        AB-          Diabetes   
7  CHrisTInA MARtinez   20  Female         A+            Cancer   
8     JASmINe aGuIlaR   82    Male        AB+            Asthma   
9    ChRISTopher BerG   58  Female        AB-            Cancer   

  Date of Admission            Doctor                     Hospital  \
0        2024-01-31     Matthew Smith              Sons and Miller   
1        2019-08-20   Samantha Davies                      Kim Inc   
2        2022-09-22  Tiffany Mitchell                     Cook PLC   
3        2020-11-18       Kevin Wells   Hernandez Rogers and Vang,   
4        2022-09-19    Kathleen Hanna                  White-White   
5        2023-12-20     Taylor Newton               Nunez-Humphrey   
6        2020-11-03       Kelly Olson              Group Middleton   
7        2021-12-28    Suzanne Thomas  Powell Robinson and Valdez,   
8        2020-07-01   Daniel Ferguson                Sons Rich and   
9        2021-05-23       Heather Day               Padilla-Walker   

  Insurance Provider  Billing Amount  Room Number Admission Type  \
0         Blue Cross    18856.281306          328         Urgent   
1           Medicare    33643.327287          265      Emergency   
2              Aetna    27955.096079          205      Emergency   
3           Medicare    37909.782410          450       Elective   
4              Aetna    14238.317814          458         Urgent   
5   UnitedHealthcare    48145.110951          389         Urgent   
6           Medicare    19580.872345          389      Emergency   
7              Cigna    45820.462722          277      Emergency   
8              Cigna    50119.222792          316       Elective   
9   UnitedHealthcare    19784.631062          249       Elective   

  Discharge Date   Medication  Test Results  
0     2024-02-02  Paracetamol        Normal  
1     2019-08-26    Ibuprofen  Inconclusive  
2     2022-10-07      Aspirin        Normal  
3     2020-12-18    Ibuprofen      Abnormal  
4     2022-10-09   Penicillin      Abnormal  
5     2023-12-24    Ibuprofen        Normal  
6     2020-11-15  Paracetamol  Inconclusive  
7     2022-01-07  Paracetamol  Inconclusive  
8     2020-07-14      Aspirin      Abnormal  
9     2021-06-22  Paracetamol  Inconclusive

In [9]:
hc_df['Date of Admission'] = pd.to_datetime(hc_df['Date of Admission'])
hc_df['Discharge Date'] = pd.to_datetime(hc_df['Discharge Date'])

In [10]:
hc_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Name                55500 non-null  object        
 1   Age                 55500 non-null  int64         
 2   Gender              55500 non-null  object        
 3   Blood Type          55500 non-null  object        
 4   Medical Condition   55500 non-null  object        
 5   Date of Admission   55500 non-null  datetime64[ns]
 6   Doctor              55500 non-null  object        
 7   Hospital            55500 non-null  object        
 8   Insurance Provider  55500 non-null  object        
 9   Billing Amount      55500 non-null  float64       
 10  Room Number         55500 non-null  int64         
 11  Admission Type      55500 non-null  object        
 12  Discharge Date      55500 non-null  datetime64[ns]
 13  Medication          55500 non-null  object    

### Data Exploration

In [11]:
hc_df['Admission Type'].value_counts()

Elective     18655
Urgent       18576
Emergency    18269
Name: Admission Type, dtype: int64

There are three categories of admission type. Class imbalance is not oberserved as each category has ~18k records.
* Elective
* Urgent
* Emergency

In [12]:
hc_df['Age'].describe(include='all')

count    55500.000000
mean        51.539459
std         19.602454
min         13.000000
25%         35.000000
50%         52.000000
75%         68.000000
max         89.000000
Name: Age, dtype: float64

The Age of patients is ranging from 13 to 89 years. 
* Plot histogram and Box plot
* Run statistical test for normal distribution

In [13]:
hc_df['Billing Amount'].describe(include='all')

count    55500.000000
mean     25539.316097
std      14211.454431
min      -2008.492140
25%      13241.224652
50%      25538.069376
75%      37820.508436
max      52764.276736
Name: Billing Amount, dtype: float64

Assuming the amount to be in USD. On an average patients are billed $25,539. 
* Plot histogram for skewness check.
* Check the -ve amount under minimum
* Find the records with -ve billing amount

In [14]:
hc_df['Blood Type'].value_counts()

A-     6969
A+     6956
AB+    6947
AB-    6945
B+     6945
B-     6944
O+     6917
O-     6877
Name: Blood Type, dtype: int64

There are eight categories of blood type. Class imbalance is not oberserved as each category has ~6k records.
* A-,A+
* B-,B+
* AB-,AB+
* O-,O+

In [15]:
hc_df['Date of Admission'].describe()

/var/folders/z1/qyczzcsn20q235vpf9n6j17w0000gn/T/ipykernel_21483/1963798732.py:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  hc_df['Date of Admission'].describe()


count                   55500
unique                   1827
top       2024-03-16 00:00:00
freq                       50
first     2019-05-08 00:00:00
last      2024-05-07 00:00:00
Name: Date of Admission, dtype: object

In [16]:
hc_df['Discharge Date'].describe(include='all')

/var/folders/z1/qyczzcsn20q235vpf9n6j17w0000gn/T/ipykernel_21483/2651826647.py:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  hc_df['Discharge Date'].describe(include='all')


count                   55500
unique                   1856
top       2020-03-15 00:00:00
freq                       53
first     2019-05-09 00:00:00
last      2024-06-06 00:00:00
Name: Discharge Date, dtype: object

Patients data ranges from May 2019 to May 2024.
* Time series analysis can be performed.
* Trend in Admission dates must be observed along side other variables.
* Time taken by patients from admission to discharge must be noted. 

In [17]:
hc_df['Doctor'].value_counts()

Michael Smith           27
Robert Smith            22
John Smith              22
Michael Johnson         20
James Smith             20
                        ..
Audrey Zimmerman DDS     1
Justin Banks             1
Joseph Williams Jr.      1
Jason Franklin           1
Jeffrey Moore            1
Name: Doctor, Length: 40341, dtype: int64

In [18]:
hc_df['Doctor'].describe(include='all')

count             55500
unique            40341
top       Michael Smith
freq                 27
Name: Doctor, dtype: object

* Among all doctors Michael Smith has attended most patients followed by Robert Smith.
* There are 40,341 unique doctors in this data.
* Top 10 Doctors in the data, top 10 doctors who attended most patients by Hospital, Gender, Diseases and more can be studied furthur. 

In [19]:
hc_df['Gender'].describe(include='all')

count     55500
unique        2
top        Male
freq      27774
Name: Gender, dtype: object

In [20]:
hc_df['Gender'].value_counts()

Male      27774
Female    27726
Name: Gender, dtype: int64

There are 2 genders Male and Female.
* Gender based studies can be sensitive.
* In scope of diseases, hospital, medication, test results analysis will be performed for futher insights and trends.

In [21]:
hc_df['Hospital'].value_counts()

LLC Smith                      44
Ltd Smith                      39
Johnson PLC                    38
Smith Ltd                      37
Smith PLC                      36
                               ..
and Montoya Flores, Boyer       1
Carter and Dunn King,           1
Hall, Brown Black and           1
Peterson Scott and Thomas,      1
Moreno Murphy, Griffith and     1
Name: Hospital, Length: 39876, dtype: int64

In [22]:
hc_df['Hospital'].describe(include='all')

count         55500
unique        39876
top       LLC Smith
freq             44
Name: Hospital, dtype: object

There are 39,876 unique hospitals in this data. LLC Smith hospital is the most visited followed by Ltd Smith.

In [23]:
hc_df['Insurance Provider'].value_counts()

Cigna               11249
Medicare            11154
UnitedHealthcare    11125
Blue Cross          11059
Aetna               10913
Name: Insurance Provider, dtype: int64

In [24]:
hc_df['Insurance Provider'].describe()

count     55500
unique        5
top       Cigna
freq      11249
Name: Insurance Provider, dtype: object

Cigna is the top insurance provider among 5 other insurance providers. The 5 insurance providers are equally distributed with ~11k among all patients.
* list the medical conditions.
* list the insurance providers

In [25]:
hc_df['Medical Condition'].value_counts()

Arthritis       9308
Diabetes        9304
Hypertension    9245
Obesity         9231
Cancer          9227
Asthma          9185
Name: Medical Condition, dtype: int64

In [26]:
hc_df['Medical Condition'].describe(include='all')

count         55500
unique            6
top       Arthritis
freq           9308
Name: Medical Condition, dtype: object

Arthritis is the top Medical condition among 6 other medical conditiona. The 6 medical conditiona are equally distributed with ~9k among all patients.
* list the medical conditions.
* list the medical conditions

In [27]:
hc_df['Medication'].value_counts()

Lipitor        11140
Ibuprofen      11127
Aspirin        11094
Paracetamol    11071
Penicillin     11068
Name: Medication, dtype: int64

In [28]:
hc_df['Medication'].describe()

count       55500
unique          5
top       Lipitor
freq        11140
Name: Medication, dtype: object

Lipitor is the top Medication among 5 other medications. The 5 medications are equally distributed with ~11k among all patients.
* list the medical conditions

In [31]:
hc_df['Room Number'] = hc_df['Room Number'].astype(str)

In [32]:
hc_df['Room Number'].describe(include='all')

count     55500
unique      400
top         393
freq        181
Name: Room Number, dtype: object

In [33]:
hc_df['Room Number'].value_counts()

393    181
491    177
104    175
420    175
209    171
      ... 
253    112
254    111
257    111
381    110
398    109
Name: Room Number, Length: 400, dtype: int64

There are 400 rooms across all the hospitals in this dataset.
Most used rooms are 393, 491, 104, 420 and 209.

* plot a graph to see the distribution

In [30]:
hc_df['Test Results'].value_counts()

Abnormal        18627
Normal          18517
Inconclusive    18356
Name: Test Results, dtype: int64

In [34]:
hc_df['Test Results'].describe()

count        55500
unique           3
top       Abnormal
freq         18627
Name: Test Results, dtype: object

There are 3 types of results Abnormal, Normal and Inconclusive.

In [35]:
hc_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Name                55500 non-null  object        
 1   Age                 55500 non-null  int64         
 2   Gender              55500 non-null  object        
 3   Blood Type          55500 non-null  object        
 4   Medical Condition   55500 non-null  object        
 5   Date of Admission   55500 non-null  datetime64[ns]
 6   Doctor              55500 non-null  object        
 7   Hospital            55500 non-null  object        
 8   Insurance Provider  55500 non-null  object        
 9   Billing Amount      55500 non-null  float64       
 10  Room Number         55500 non-null  object        
 11  Admission Type      55500 non-null  object        
 12  Discharge Date      55500 non-null  datetime64[ns]
 13  Medication          55500 non-null  object    

### Analysis to gain further insights from the data.

* Creation of additional variables.

In [48]:
hc_df['Days in bed'] = hc_df['Discharge Date'] - hc_df['Date of Admission']

In [50]:
hc_df['Days in bed'].head(10)

0    2 days
1    6 days
2   15 days
3   30 days
4   20 days
5    4 days
6   12 days
7   10 days
8   13 days
9   30 days
Name: Days in bed, dtype: timedelta64[ns]

In [53]:
hc_df['Days in bed'] = hc_df['Days in bed'].astype('timedelta64[D]').astype(np.int64)


In [54]:
hc_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Name                55500 non-null  object        
 1   Age                 55500 non-null  int64         
 2   Gender              55500 non-null  object        
 3   Blood Type          55500 non-null  object        
 4   Medical Condition   55500 non-null  object        
 5   Date of Admission   55500 non-null  datetime64[ns]
 6   Doctor              55500 non-null  object        
 7   Hospital            55500 non-null  object        
 8   Insurance Provider  55500 non-null  object        
 9   Billing Amount      55500 non-null  float64       
 10  Room Number         55500 non-null  object        
 11  Admission Type      55500 non-null  object        
 12  Discharge Date      55500 non-null  datetime64[ns]
 13  Medication          55500 non-null  object    

In [ ]:
hc_df['Age Group'] = 